implementation/operations.go

```diff
@@ -0,0 +1,144 @@
+package implementation
+
+import (
+	"errors"
+	"github.com/aditya/ProjectCatalog/models"
+	"sort"
+	"strconv"
+	"time"
+)
+
+type Inmemoryimplement struct{
+	ProductArr []models.Product
+
+}
+
+type TimeStampStructure struct{
+	prodId int64
+	timeInSec int64
+}
+
+var mp map[TimeStampStructure]int64
+
+func Initializer(){
+	mp=make(map[TimeStampStructure]int64)
+	return
+}
+
+func min(a int64, b int64) int64{
+	if a<=b{
+		return a
+	}
+
+	return b
+
+}
+
+func (h *Inmemoryimplement) CreateProduct(product models.Product) error{
+
+	for _, prod:=range h.ProductArr{
+		if prod.Name==product.Name{
+			return errors.New("Product already exist.")
+		}
+	}
+	product.ID=strconv.Itoa(len(h.ProductArr)+1)
+	h.ProductArr = append(h.ProductArr,product)
+
+	return nil
+
+}
+
+func (h *Inmemoryimplement) ShowProduct() []models.Product{
+	return h.ProductArr
+}
+
+func (h *Inmemoryimplement) ShowProductById(productId string) models.Product{
+	for _,product:=range h.ProductArr{
+		if product.ID==productId{
+			return product
+		}
+	}
+	return models.Product{}
+}
+
+func (h *Inmemoryimplement) UpdateProduct(updatedProduct models.Product,productName string) error{
+	for i,item:=range h.ProductArr{
+		if item.Name==productName{
+			updatedProduct.ID = strconv.Itoa(i + 1)
+			if item.Price != updatedProduct.Price {
+				item.Price = updatedProduct.Price
+			}
+			if item.Description != updatedProduct.Description {
+				item.Description = updatedProduct.Description
+			}
+			h.ProductArr[i] = updatedProduct
+			return nil
+		}
+	}
+	return errors.New("Product do not exist.")
+}
+
+func (h *Inmemoryimplement) BuyProduct(purchaseProduct models.Product, productName string) error{
```

Comment:
what will happen when multiple users try to buy the same product at the same time?